src/types/mod.rs

```diff
@@ -815,6 +815,16 @@ pub trait ToSql: fmt::Debug {
                       -> Result<IsNull>;
 }
 
+impl<'a, T> ToSql for &'a T where T: ToSql {
+    to_sql_checked!();
```

Comment:
I believe all you need to do to fix the ICE is to implement this manually - see the implementations for &str and &[u8] for example.